platforms/jvm/toolchains-jvm/src/testFixtures/groovy/org/gradle/jvm/toolchain/JdkRepository.groovy

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain
+
+import org.apache.commons.compress.archivers.ArchiveOutputStream
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream
+import org.apache.commons.compress.utils.IOUtils
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.nativeintegration.services.FileSystems
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.gradle.util.internal.RelativePathUtil
+
+import java.nio.file.Files
+
+
+class JdkRepository {
+
+    private BlockingHttpServer server
+
+    private Jvm jdk
+
+    private String archiveName
+
+    JdkRepository(JavaVersion javaVersion) {
+        this(Objects.requireNonNull(AvailableJavaHomes.getJdk(javaVersion)), "jdk.zip")
+    }
+
+    JdkRepository(Jvm jdk, String archiveName) {
+        this.jdk = jdk
+
+        this.archiveName = archiveName
+
+        server = new BlockingHttpServer(1000)
+
+        expectHead()
+        expectGet()
+    }
+
+    URI start() {
+        server.start()
+        server.uri(archiveName)
+    }
+
+    void expectHead() {
+        server.expect(server.head(archiveName))
+    }
+
+    void expectGet() {
+        server.expect(server.get(archiveName, { e ->
+            e.sendResponseHeaders(200, 0)
+            zip(jdk.javaHome, e.getResponseBody())
```

Comment:
❓ Does that mean that each test using a shared `JdkRepository` will create a new ZIP of a JDK? Can't we reuse these ZIPs somehow to make the tests faster?

Reply:
I've made the change, but I'm not sure it helps a lot. I have to save the archive to a temporary file because it's too much to keep in memory, and then loading from that file doesn't seem to be much faster than re-compressing it. But it's not worse, so let's keep it.